File: php/practice/classic_puzzle/easy/the-descent.php
<?php
/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/


// game loop
while (TRUE)
{
    $target_h = 0;
    $target_id = 0;

    for ($i = 0; $i < 8; $i++)
    {
        fscanf(STDIN, "%d",
            $mountainH // represents the height of one mountain.
        );

        if ($target_h < $mountainH) {
            $target_h  = $mountainH;
            $target_id = $i;
        }
    }

    // Write an action using echo(). DON'T FORGET THE TRAILING \n
    // To debug (equivalent to var_dump): error_log(var_export($var, true));

    // echo("4\n"); // The index of the mountain to fire on.

    echo $target_id."\n";
}

?>

File: go/practice/classic_puzzle/easy/the-descent.go
package main

import "fmt"

/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/

func main() {
    for {
        var mountainHeights []int
        for i := 0; i < 8; i++ {
            // mountainH: represents the height of one mountain.
            var mountainH int
            fmt.Scan(&mountainH)

            mountainHeights = append(mountainHeights, mountainH)
        }

        targetHeight := 0
        targetId := 0
        for i, v := range mountainHeights {
            if targetHeight < v {
                targetHeight = v
                targetId = i
            }
        }

        // fmt.Fprintln(os.Stderr, "Debug messages...")
        // fmt.Println("4") // The index of the mountain to fire on.
        fmt.Println(targetId)
    }
}

